core/dbt/contracts/graph/manifest.py

```diff
@@ -671,18 +664,22 @@ def build_disabled_by_file_id(self):
                 disabled_by_file_id[node.file_id] = node
         return disabled_by_file_id
 
+    def _get_parent_adapter_types(self, adapter_type: str) -> List[str]:
+        from dbt.adapters.factory import get_adapter_type_names
+
+        # order matters for dispatch:
+        #  1. current adapter
+        #  2. any parent adapters (dependencies)
+        #  3. 'default'
+        return get_adapter_type_names(adapter_type) + ["default"]
+
     def _materialization_candidates_for(
         self,
         project_name: str,
         materialization_name: str,
-        adapter_type: Optional[str],
+        adapter_type: str,
+        specificity: int,
```

Comment:
@volkangurel I might also leave that specificity comment here since this seems to be where it is passed in.

Reply:
Done!